clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -6387,6 +6395,78 @@ static void handleIntelFPGAMaxReplicatesAttr(Sema &S, Decl *D,
   S.AddIntelFPGAMaxReplicatesAttr(D, A, A.getArgAsExpr(0));
 }
 
+// Handle [[intel:fpga_pipeline]] attribute.
+void Sema::AddSYCLIntelFpgaPipelineAttr(Decl *D, const AttributeCommonInfo &CI,
+                                        Expr *E) {
+  if (!E->isValueDependent()) {
+    // Validate that we have an integer constant expression and then store the
+    // converted constant expression into the semantic attribute so that we
+    // don't have to evaluate it again later.
+    llvm::APSInt ArgVal;
+    ExprResult Res = VerifyIntegerConstantExpression(E, &ArgVal);
+    if (Res.isInvalid())
+      return;
+    E = Res.get();
+
+    // This attribute requires a non-negative value.
+    if (ArgVal < 0) {
+      Diag(E->getExprLoc(), diag::err_attribute_requires_positive_integer)
+          << CI << /*non-negative*/ 1;
+      return;
+    }
+
+    // Check to see if there's a duplicate attribute with different values
+    // already applied to the declaration.
+    if (const auto *DeclAttr = D->getAttr<SYCLIntelFpgaPipelineAttr>()) {
+      // If the other attribute argument is instantiation dependent, we won't
+      // have converted it to a constant expression yet and thus we test
+      // whether this is a null pointer.
+      if (const auto *DeclExpr = dyn_cast<ConstantExpr>(DeclAttr->getValue())) {
+        if (ArgVal != DeclExpr->getResultAsAPSInt()) {
```

Comment:
So this is a bit of an interesting case -- given that this attribute is effectively boolean, there's no notional difference between `[[intel::fpga_pipeline(1)]]` and `[[intel::fpga_pipeline(2)]]`. I think that still is worth diagnosing though, as the user is likely confused as to what's going on if they do this sort of thing. WDYT?